pennylane_sf/vgbs.py

```diff
@@ -0,0 +1,142 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Strawberry Fields variational GBS device
+========================================
+
+**Module name:** :mod:`pennylane_sf.vgbs`
+
+.. currentmodule:: pennylane_sf.vgbs
+
+TODO
+
+Classes
+-------
+
+.. autosummary::
+   StrawberryFieldsVGBS
+
+
+Code details
+~~~~~~~~~~~~
+"""
+from collections import OrderedDict
+from thewalrus.quantum import find_scaling_adjacency_matrix as rescale
+import pennylane as qml
+
+import numpy as np
+
+import strawberryfields as sf
+from strawberryfields.utils import all_fock_probs_pnr
+
+# import gates
+from strawberryfields.ops import Dgate, GraphEmbed, MeasureFock
+
+from .expectations import identity
+from .simulator import StrawberryFieldsSimulator
+
+
+class GraphTrain(qml.operation.CVOperation):
+    """TODO"""
+    do_check_domain = False # turn off parameter domain checking
```

Comment:
By setting this class attribute, we can turn off the operation parameter validation 🙂 Avoids having to overwrite `__init__`, and allows each parameter to be a different type.

Reply:
Excellent, looks much better!